pkg/asset/machines/master.go

```diff
@@ -263,6 +263,11 @@ func (m *Master) Generate(dependencies asset.Parents) error {
 		}
 		pool.Platform.Azure = &mpool
 
+		if mpool.OSDisk.DiskType == "Standard_LRS" {
```

Comment:
This validation is only for the master. The worker will still take up the value. Since the value for the master alone cannot be HDD but it can be for the workers and since there is only one diskType for both master and worker, I thought placing it here would minimize code.

Reply:
See https://github.com/openshift/installer/pull/3520#discussion_r417025533

The code in this file is for creating machine assets. The code in the link above is where we validate install configs